### PR TITLE
doc: add Refine tutorial

### PR DESCRIPTION
**Pull request by necatiozmen**
## What does this PR do?

This PR adds a tutorial and quick start guide to the website for Refine and Appwrite. It was done with the knowledge of @gewenyu99 

## Test Plan
I was not able to render Refine icons after adding them the project. I may need a help for this.

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

Yes

### CODE REVIEW

**Comment by TGlide** [comment]
Hi! The workflow for icons is: stick the SVG in `src/icons/svg`, then run `pnpm run icons:generate`.

File: src/icons/output/aw-icon.svg
Comment (gewenyu99) [inline src/icons/output/aw-icon.svg:1]: Take a look at Thomas' comments for the icon workflow. Sorry this isn't documented anywhere! First time we're doing an integration here with the new docs.

**Review by gewenyu99 — CHANGES_REQUESTED** [review]

**Comment by necatiozmen** [comment]
> Hi! The workflow for icons is: stick the SVG in `src/icons/svg`, then run `pnpm run icons:generate`.

Hi @TGlide,

I've placed the SVG in the `src/icons/svg` directory and executed the `pnpm run icons:generate` command. You can see the commit here: https://github.com/appwrite/website/pull/350/commits/6ea7d1fa0885795625da6a610d6a5f41b7e27af9.

However, I'm encountering an issue where the Refine SVG isn't appearing on the tutorials page. Interestingly, it displays correctly on both the quick start and home pages.

File: src/routes/docs/tutorials/+page.svelte [hunk 1]
```diff
@@ -19,7 +19,8 @@
         sveltekit: 'icon-svelte',
         android: 'icon-android',
         apple: 'icon-apple',
-        flutter: 'icon-flutter'
+        flutter: 'icon-flutter',
+        refine: 'icon-refine'
```
Comment (gewenyu99) [inline src/routes/docs/tutorials/+page.svelte:23]: Try `aw-icon-refine`?

Comment (necatiozmen) [reply src/routes/docs/tutorials/+page.svelte]: Thanks, it worked. The PR is ready now.

**Comment by gewenyu99** [comment]
Screenshot 2023-11-28 at 9 28 35 AM
I wonder if this logo looks like it's the right size. Let me check with design.

File: src/routes/docs/tutorials/refine/step-2/+page.markdoc [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+```sh
+npm create refine-app@latest -- --preset refine-appwrite
+```
+
+We're using the `refine-appwrite` preset that installs the [`@refinedev/appwrite`](https://github.com/refinedev/refine/tree/master/packages/appwrite) which already has the  Appwrite dependency pre-configured.
+
+To make this example more visual, we'll use the Ant Desing UI package which natively supported by Refine.
+
+{% info title="Note" %}
```
Comment (gewenyu99) [inline src/routes/docs/tutorials/refine/step-2/+page.markdoc:20]:
```suggestion
{% info title="Existing projects" %}
```

**Comment by gewenyu99** [comment]
refine

**Comment by necatiozmen** [comment]
> refine
> 
> We added the same padding size for the refine icon as we do all our other icons in the aw-icons library, so they'll match in size :)

Thank you, I updated it and renders properly now.

File: src/routes/docs/tutorials/refine/step-4/+page.markdoc [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+} from '@refinedev/react-router-v6';
+import '@refinedev/antd/dist/reset.css';
+
+import { App as AntdApp, ConfigProvider } from 'antd';
+import { BrowserRouter, Outlet, Route, Routes } from 'react-router-dom';
+
+import { appwriteClient } from './utility';
+import { authProvider } from './authProvider';
+
+const App: React.FC = () => {
```
[the remark this answers is not in the document]
Comment (necatiozmen) [reply src/routes/docs/tutorials/refine/step-4/+page.markdoc:85]: I'll update all code blocks indentations.

File: src/routes/docs/tutorials/refine/step-6/+page.markdoc [hunk 4]
```diff
@@ -0,0 +1,396 @@
...
+import { App as AntdApp, ConfigProvider } from 'antd';
+import { BrowserRouter, Outlet, Route, Routes } from 'react-router-dom';
+
+import { appwriteClient } from './utility';
+import { authProvider } from './authProvider';
+import { PostCreate, PostEdit, PostList, PostShow } from './pages/posts';
+
+const App: React.FC = () => {
+    return (
+        <BrowserRouter>
```
[the remark this answers is not in the document]
Comment (necatiozmen) [reply src/routes/docs/tutorials/refine/step-6/+page.markdoc:280]: I've updated the indentations, I think it more readable now. Please review again.

**Comment by gewenyu99** [comment]
Hey, I'm gonna merge this into a branch and carry it forward myself. Thank you so much for your help! This is looking great!

**Review by gewenyu99 — APPROVED** [review]

**Comment by necatiozmen** [comment]
> Hey, I'm gonna merge this into a branch and carry it forward myself. Thank you so much for your help! This is looking great!

Thanks you so much

**Comment by gewenyu99** [comment]
Screenshot 2023-12-13 at 3 08 26 PM
For the record.

**Pull request merged**